### DISCUSSION TITLE
How would you get the width of the window size from NvimTree?

### DISCUSSION THREAD

**Discussion by DennisTheMenace780** [nvim-tree/nvim-tree.lua#1813] [Q&A]
I have Toggle Term and Nvim Tree set up and what I find happening is that the window for NvimTree can be _above_ Toggle Term, or it can be to the left of Toggle Term.

I think that I can probably create a function for Toggle Term that, when opened, inspects the size of NvimTree and then adjusts the dimensions accordingly. I'm still pretty new to the ecosystem though, so it's not totally clear to me how I would get this information from the nvim-tree plugin over into my custom function. If someone is more familiar with the code base for NvimTree, could you tell me where I can look to get this dimension information? 

I have tried to do a `:lua print(winwidth(0))` as I understood this to return dimensions of the currently active window, but I keep getting a nil message back. 

I have taken some images below that show what I want and what I don't want to have happen. 

[What I don't want](https://freeimage.host/i/HCPWNqB) [What I do want](https://freeimage.host/i/HCPWjsV)

**Comment by alex-courtis** [comment]
winwidth is vimscript only. See `:help nvim_win_get_width()` for the lua version.

`:lua print(vim.api.nvim_win_get_width(0))` prints the current window width.
`:lua print(vim.api.nvim_win_get_width(1002))` for a specific winid.

You can use an internal function to determine whether a buffer is nvim-tree:
`:lua print(require("nvim-tree.utils").is_nvim_tree_buf(2))`.
You can glue that together by enumerating windows `vim.api.nvim_list_wins` and their buffers `vim.api.nvim_win_get_buf`
